htmd/parameterization/util.py

```diff
@@ -383,6 +383,72 @@ def makeAtomNamesUnique(mol):
 
     return mol
 
+def detectChiralCenters(mol):
+    """
+    Detect chiral centers
+
+    Parameters
+    ----------
+    mol: Molecule
+        Molecule to detect chiral centers
+
+    Return
+    ------
+    results: list of list of tuples
+        List of list of chircal centers for each frame.
+        The chiral centers are tuples made of an atom index and a label ('S', 'R', '?').
+
+    Examples
+    --------
+    >>> from htmd.home import home
+    >>> from htmd.molecule.molecule import Molecule
+
+    >>> molFile = os.path.join(home('test-param'), 'H2O2.mol2')
+    >>> mol = Molecule(molFile)
+    >>> detectChiralCenters(mol)
+    [[]]
+
+    >>> molFile = os.path.join(home('test-param'), 'fluorchlorcyclopronol.mol2')
+    >>> mol = Molecule(molFile)
+    >>> detectChiralCenters(mol)
+    [[(0, 'R'), (2, 'S'), (4, 'R')]]
+
+    >>> mol.appendFrames(mol)
+    >>> detectChiralCenters(mol)
+    [[(0, 'R'), (2, 'S'), (4, 'R')], [(0, 'R'), (2, 'S'), (4, 'R')]]
+    """
+
+    from htmd.molecule.molecule import Molecule
+    from rdkit.Chem import MolFromMol2File, AssignAtomChiralTagsFromStructure, FindMolChiralCenters
+
+    if not isinstance(mol, Molecule):
+        raise TypeError('"mol" has to be instance of {}'.format(Molecule))
+
+    all_chiral_centers = []
+
+    for i in range(mol.numFrames):
+
+        # Copy the molecule and drop unwanted frames
+        htmd_mol = mol.copy()
+        htmd_mol.dropFrames(keep=i)
+
+        # Set atom types to elements, overwise rdkit refuse to read a MOL2 file
+        htmd_mol.atomtype = htmd_mol.element
+
+        # Convert Molecule to rdkit Mol
+        with TemporaryDirectory() as tmpDir:
+            filename = os.path.join(tmpDir, 'mol.mol2')
+            htmd_mol.write(filename)
+            rdkit_mol = MolFromMol2File(filename, removeHs=False)
+        assert mol.numAtoms == rdkit_mol.GetNumAtoms()
+
+        # Detect chiral centers and assign their labels
+        AssignAtomChiralTagsFromStructure(rdkit_mol)
+        chiral_centers = FindMolChiralCenters(rdkit_mol, includeUnassigned=True)
+
+        all_chiral_centers.append(chiral_centers)
```

Comment:
Why would you want to check all frames of a molecule for chiral centers? I would rather make the function work on the first frame only like most molecule utils. From what I see you only use the first frame anyway.

Reply:
I agree, this could be simplified!